src/main/java/ch/njol/skript/conditions/CondIsOnline.java

```diff
@@ -29,36 +29,43 @@
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.util.Kleenean;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Is Online")
-@Description("Checks whether a player is online.")
-@Examples({"player is online",
-		"player-argument is offline"})
+@Description("Checks whether a player is online. The 'connected' pattern will return false once this player leaves the server, even if they rejoin.")
```

Comment:
The description needs to mention that since Skript has a "player variable fix" config option that replaces old offlineplayer references in vars with new references, this expression may not work quite right when used with variables.

For example, the below loop with continue even after rejoining, if you do it within the 2 seconds. This does not occur if that setting is turned off or if you use an expression for the player.
```
on join:
    add 1 to {test}
    set {_test} to {test}
    set {_player} to player
    while {_player} is connected:
        broadcast "loop %{_test}%"
        wait 2 seconds
```

Reply:
I'm going to be honest I don't know what any of that means 😬

Reply:
Since skript updates the player variable reference a variable set to a player will always be connected compared to `player is connected` not being an updated player instance.

Reply:
> I'm going to be honest I don't know what any of that means 😬

Basically just add `"This condition may not work as expected if used with a variable rather than a player expression like event-player"` or something similar to the description.
Or just prevent variables from being used here, either or.

Reply:
Ooh can you do that suggest changes thingy so i can just click a button!

Reply:
> Ooh can you do that suggest changes thingy so i can just click a button!

sure, for the description change or forbidding variables?

Reply:
Probably yes